pkg/controller/bookkeepercluster/bookie.go

```diff
@@ -365,29 +365,101 @@ func createVolumeMount(ledgerDirs []string, journalDirs []string, indexDirs []st
 }
 
 func makeBookieVolumeClaimTemplates(bk *v1alpha1.BookkeeperCluster) []corev1.PersistentVolumeClaim {
-	return []corev1.PersistentVolumeClaim{
-		{
+	var ledgerDirs, journalDirs, indexDirs []string
+	var ok bool
+
+	if _, ok = bk.Spec.Options["ledgerDirectories"]; ok {
+		ledgerDirs = strings.Split(bk.Spec.Options["ledgerDirectories"], ",")
+	} else {
+		// default value if user did not set ledgerDirectories in options
+		ledgerDirs = append(ledgerDirs, "/bk/ledgers")
+	}
+
+	if _, ok = bk.Spec.Options["journalDirectories"]; ok {
+		journalDirs = strings.Split(bk.Spec.Options["journalDirectories"], ",")
+	} else {
+		// default value if user did not set journalDirectories in options
+		journalDirs = append(journalDirs, "/bk/journal")
+	}
+
+	if _, ok = bk.Spec.Options["indexDirectories"]; ok {
+		indexDirs = strings.Split(bk.Spec.Options["indexDirectories"], ",")
+	} else {
+		// default value if user did not set indexDirectories in options
+		indexDirs = append(indexDirs, "/bk/index")
+	}
+
+	var claims []corev1.PersistentVolumeClaim
+	if len(ledgerDirs) > 1 {
+		for i := 0; i < len(ledgerDirs); i++ {
+			name := LedgerDiskName + strconv.Itoa(i)
+			pvc := corev1.PersistentVolumeClaim{
```

Comment:
This is changing the current behavior of having one PVC with multiple mount points. We should make this option configurable.  for multiple PVCs  and we should make the default as one PVC with multiple mount points.

Reply:
> This is changing the current behavior of having one PVC with multiple mount points. We should make this option configurable. for multiple PVCs and we should make the default as one PVC with multiple mount points.

OK. We will change the code to use a bk-op option, temporarily named as multiVolumesSetPerBookieEnabled, as a flag to control this behavior. By default or when it's set to false, the original one PVC with multiple mount points will be used. The multiple-PVC support will only be used when multiVolumesSetPerBookieEnabled is set to true.